rust-connectors/sinks/sql/src/main.rs

```diff
@@ -46,11 +46,27 @@ async fn main() -> anyhow::Result<()> {
     );
     while let Some(Ok(consumer_record)) = stream.next().await {
         let record: Record = consumer_record.into_inner();
-        let output = transformations.transform(record)?;
+        let output = match transformations.transform(record) {
```

Comment:
This may lead to discrepancies between the data in the fluvio topic and the SQL table. Until we have some ability to configure the level of consistency to use, we should just fail fast once the first error occurs.

Reply:
Part of the issue here is that if it fails, the pod restarts and we'll end up with the same result I think.

Reply:
True, but it is only because we don't have offset management yet. Once offset is stored on the SPU side, we should not have gaps or duplicates. The processing can be stuck forever, but it is another story.

Reply:
ah true

Reply:
I think I'd argue this should be a configurable behavior, common to all connectors, and something we require the user to choose during connector creation. Something like `ErrorPolicy: Halt | Discard | Retry`. A crash (or just a hard process stop) is going to be trickier for a user to diagnose, especially as the pods restart and recycle logs, and it will contribute to a sense that connectors are flaky or unstable, even if it's desirable at some level.